index.js

```diff
@@ -0,0 +1,63 @@
+const crypto = require('crypto');
+const authMiddleware = require('express-basic-auth-safe');
+const debug = require('diagnostics')('authboot');
+
+module.exports = function (opts) {
+  opts = opts || {}
+  return function (app, options, callback) {
+    const users = new Map(
+      Object.entries(opts.users
+        || options.users
+        || app.config.get('auth:users')
+        || {}
+      )
+    );
+
+    const realm = opts.realm
+      || options.realm
+      || app.config.get('auth:realm');
+
+    const challenge = opts.challenge
+      || options.challenge
+      || app.config.get('auth:challenge');
+
+    const unauthorizedResponse = opts.errorPayload
+      || options.errorPayload
+      || app.config.get('auth:erroPayload')
+      || { error: 'Not authorized' }
+
+    const lookupOpt = typeof opts.lookup === 'function' ? opts.lookup : null;
+
+    if (challenge && !realm) {
+      return callback(new Error('authboot requires a specified realm if a challenge request will be sent.'));
+    }
+
+    app.authboot = {};
+    const lookup = app.authboot.lookup = (lookupOpt || function ({ name, password }, callback) {
```

Comment:
is there a reason to store this function on authboot? whats the expectation? Is that just for testing?

Reply:
its more or less for convenience for also integrating with the npm auth strategy to use the same lookup function.